src/Renderer/QueryRenderer.ts

```diff
@@ -283,6 +283,7 @@ class QueryRenderChild extends MarkdownRenderChild {
     }
 
     private addEditButton(listItem: HTMLElement, task: Task, allTasks: Task[]) {
+        // TODO Need to figure out how to replace this with a call to createElement() and then set cls, href and title.
         const editTaskPencil = listItem.createEl('a', {
             cls: 'tasks-edit',
             href: '#',
```

Comment:
There are three remaining calls to `element.createEl()` that need to be converted to `createElement()`. I was not able to easily figure out how to convert their additional options, like classes...

I think the options I added to the `createElement()` call are the wrong type of options. They provide access to some `is` option - see the link in the JSDocs for the new `createElement()`.

There are so few few calls remaining to the old `element.createEl()` - could they just be changed to add the class, href etc manually????

Reply:
The `is` option is used for custom components' definition (that is described in the link you mentioned), so I would remove the options from our function.

We can use some TypeScript features to set these options with the current interface (`{ cls: '...', href: '#', ...}`), but I think it is just simpler for now to separate the element creation and its options. For this case the options are set with the following code:

```
editTaskPencil.addClass('tasks-edit');
editTaskPencil.href = '#';
editTaskPencil.title = 'Edit task';
```

This _shall_ work, I didn't test it but you can easily see whether this breaks the outcome or not.

Reply:
Thank you very much. This worked well, I just had to reorder the calls slightly to get exactly unchanged HTML.